jq/jv.go

```diff
@@ -231,9 +332,25 @@ func (jv *Jv) ToGoVal() interface{} {
 	case C.JV_KIND_STRING:
 		return jv._string()
 	case C.JV_KIND_ARRAY:
-		fallthrough
+		len := jv.Copy().ArrayLength()
+		ary := make([]interface{}, len)
+		for i := 0; i < len; i++ {
+			v := jv.Copy().ArrayGet(i)
+			ary[i] = v.ToGoVal()
+			v.Free()
+		}
+		return ary
 	case C.JV_KIND_OBJECT:
-		panic(fmt.Sprintf("ToGoVal not implemented for %#v", kind))
+		obj := make(map[string]interface{})
+		for iter := C.jv_object_iter(jv.jv); C.jv_object_iter_valid(jv.jv, iter) != 0; iter = C.jv_object_iter_next(jv.jv, iter) {
```

Comment:
Does jv_object_iter assume ownership of this jv? If so then we need a Copy() before hand, otherwise this case would consume the invocant but the other's don't.

Reply:
I think from looking at the usage of jv_object_iter in jv_print.c (line 258) it doesn't, as that branch doesn't have a `jv_copy(x)`  but other use of x in that function do have a copy.

Does that sound right?

Reply:
I believe it doesn't. If you look at `jv_object_iter`, it's just a special case of `jv_object_iter_next`, and it would be crazy for that one to consume a reference every time it's run.